src/test/java/com/exasol/adapter/dynamodb/documentpath/ArrayLookupPathSegmentTest.java

```diff
@@ -0,0 +1,34 @@
+package com.exasol.adapter.dynamodb.documentpath;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+
+import org.junit.jupiter.api.Test;
+
+class ArrayLookupPathSegmentTest {
+    private static final ArrayLookupPathSegment TEST_SEGMENT = new ArrayLookupPathSegment(1);
+    private static final ArrayLookupPathSegment EQUAL_SEGMENT = new ArrayLookupPathSegment(1);
+    private static final ArrayLookupPathSegment UNEQUAL_SEGMENT = new ArrayLookupPathSegment(2);
+
+    @Test
+    void testEquality() {
+        assertThat(TEST_SEGMENT, equalTo(EQUAL_SEGMENT));
+    }
+
+    @Test
+    void testInequality() {
+        assertThat(TEST_SEGMENT, not(equalTo(UNEQUAL_SEGMENT)));
+    }
+
+    @Test
+    void testEqualHash() {
+        assertThat(TEST_SEGMENT.hashCode(), equalTo(EQUAL_SEGMENT.hashCode()));
+    }
+
+    @Test
+    void testUnequalHash() {
+        assertThat(TEST_SEGMENT.hashCode(), not(equalTo(UNEQUAL_SEGMENT.hashCode())));
+    }
+
+}
```

Comment:
These two tests look like the classes they test are duplication

Reply:
Even so the two classes are only a little different, they represent completely different things. I want two different classes in order to use the visitor pattern. Also the tests are different even so they look quire similar...